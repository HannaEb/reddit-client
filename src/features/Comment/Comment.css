.comment {
    background: #F8FAFB;
    border-radius: 10px;
    padding: 0.5em;
}

.comment-refs {
    display: flex;
    justify-content: space-between;
    font-size: 0.7em;
}

.comment-author {
    font-weight: bold;
}

.comment-created {
    color: #808080;
}

.comment-text {
    font-size: 0.8em;
    padding-top: 0.2em;
}
