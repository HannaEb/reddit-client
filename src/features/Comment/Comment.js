import React from 'react'
import './Comment.css'
import TimeAgo from 'react-timeago'

const Comment = (props) => {
    const { comment } = props

    return (
        <div key={comment.id} className="comment">
            <div className="comment-refs">
                {/* <p className="comment-author">{comment.author}</p> */}
                <p className="comment-created"><TimeAgo date={comment.created_utc * 1000} /></p>
            </div>
            <div className="comment-body">
                {/* <p>{comment.body}</p> */}
            </div>
        </div>
    )
}

export default Comment;
