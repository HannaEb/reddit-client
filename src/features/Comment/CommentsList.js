import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import "./Comment.css";
import Comment from "./Comment";
import { fetchComments, selectComments } from "../../slices/commentSlice";

const CommentsList = () => {
  const selectedPermalink = useSelector(
    (state) => state.comments.selectedPermalink
  );
  const comments = useSelector(selectComments);
  const status = useSelector((state) => state.comments.status);
  const error = useSelector((state) => state.comments.error);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchComments(selectedPermalink));
  }, [selectedPermalink, dispatch]);

  if (status === "loading") {
    return <div className="notification">Loading...</div>;
  }

  if (status === "failed") {
    return <div className="notification">{error}</div>;
  }

  return (
    <>
      {comments.map((comment) => (
        <Comment key={comment.id} comment={comment} />
      ))}
    </>
  );
};

export default CommentsList;
