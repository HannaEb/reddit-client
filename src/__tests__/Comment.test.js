import React from 'react';
import Comment from '../features/Comment/Comment';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';

describe('Comment component', () => {
    it('should render with props', () => {
        const wrapper = shallow(
            <Comment comment={{
                id: '1',
                author: 'Author',
                created_utc: 'Date',
                body: 'Content'
            }} />
        )
        expect(wrapper).toMatchSnapshot();
    });
});
