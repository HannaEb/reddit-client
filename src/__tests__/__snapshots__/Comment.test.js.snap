// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment component should render with props 1`] = `ShallowWrapper {}`;
